zone/spells.cpp

```diff
@@ -6770,3 +6776,24 @@ void Mob::SetHP(int64 hp)
 
 	current_hp = hp;
 }
+
+bool Mob::CheckWaterLoS(Mob* attacker, Mob* target) // checks if both attacker and target are both in or out of the water
+{
+	if (!RuleB(Spells, WaterMatchRequiredForLoS)) { // if rule is set to false, bypass check
+		return true;
+	}
+	else {
```

Comment:
The else branch can be simplified to `return Inliquid(attacker) != Inliquid(target)`

Reply:
Ah, appreciate the tip. I'll push an update late today when I get home from work. Going to add checks for bots as well.